src/java/org/apache/cassandra/repair/RepairJobDesc.java

```diff
@@ -61,6 +68,17 @@ public RepairJobDesc(UUID parentSessionId, UUID sessionId, String keyspace, Stri
         this.ranges = ranges;
     }
 
+    public UUID determanisticId()
```

Comment:
the idea is that streaming done by repair will use the RepairJob id (as defined by CASSANDRA-15399), so if a repair exists with the same planId, then this stream can map to the repair without any networking changes